### PR TITLE
Adding anchor js functionality to docs page layout

### PR DESCRIPTION
**Pull request by line47**
- Targeting h1, h2, h3 tags that are on the docs page layouts only
- Utilizing the anchor script is optional 
- Adding main.js file to set up script
- Updating README to remove confusing line in script setup

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -310,7 +310,6 @@ layout or page's [front matter] YAML in the following form:
 
 ```yml
 scripts:
-  - /path/to/script.js
```
Comment (hursey013) [inline README.md]: I think the intent here is that you can use either syntax, just the path, or the more detailed approach where you specify the `src` and `async` option.

File: README.md [hunk 2]
```diff
@@ -310,7 +310,6 @@ layout or page's [front matter] YAML in the following form:
...
   - href: /path/to/script.js
```
Comment (hursey013) [inline README.md]: Just realized this should be `src` not `href`

**Comment by hursey013** [comment]
Hey @line47, I added another approach the lets folks toggle AnchorJS on and off by using a variable in `_config.yml`. When you have a second, let me know what you think.

File: assets/js/main.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+---
+
+{% if site.anchor_js %}
+  // Adding anchor links to headers on docs layouts only
+  anchors.add('.usa-layout-docs h1, .usa-layout-docs h2, .usa-layout-docs h3');
```
Comment (line47) [inline assets/js/main.js]: I wonder if we should move the h1, h2, h3 setup into a config?

File: README.md [hunk 4]
```diff
@@ -311,7 +308,7 @@ layout or page's [front matter] YAML in the following form:
 ```yml
 scripts:
   - /path/to/script.js
-  - href: /path/to/script.js
+  - src: /path/to/script.js
```
Comment (line47) [inline README.md]: I'm wondering if we should do some explaining on how the setup in the config works in the README?

Comment (hursey013) [reply README.md]: Activating AnchorJS won't require modifying the scripts array, but if you just mean how adding custom scripts works in general I think it's already pretty well covered: https://github.com/18F/uswds-jekyll#scripts

It think adding a blurb about AnchorJS (similar to where we explain Google Analytics and DAP) would be good. Basically just what it is, and how to activate it.

**Comment by line47** [comment]
I was looking at something like this in the `_config`
```yml
anchor_js: true
anchor_js_headings:
  - h1
  - h2
  - h3
```

And something like this in the `main.js`
```js
{% if site.anchor_js %}
// Adding anchor links to headers on docs layouts only
  anchors.add('
  {% for item in site.anchor_js_headings %}
    {% if forloop.last == false %}
      .usa-layout-docs {{ item }},
    {% else %}
      .usa-layout-docs {{ item }}
    {% endif %}
  {% endfor %}');
{% endif %}
```

**Comment by hursey013** [comment]
Funny, two different approaches, same result. I pushed a commit after I read your original comment doing the same thing, albeit maybe a little over-engineered: https://github.com/18F/uswds-jekyll/pull/45/commits/61575a9d78bdbc09972aca0fd8919d04adf27b72

You have a preference on which approach to use?

**Comment by line47** [comment]
Maybe we could get some other 👀 on it to see if there is a preference ?

**Comment by afeld** [comment]
Is the question about whether the user can specify which headings get matched or not? I would only expect to use it if there were headings like the title of the page which I didn't want to be matched, but that probably wouldn't be an issue anyway since the layouts are predefined. In other words, I think it's probably overkill, but doesn't hurt anything.

**Comment by line47** [comment]
Exactly - allowing for the user to change the header targets via a config

**Comment by line47** [comment]
I think using your method sounds good @hursey013 🎉

**Pull request merged**